platform/core/src/utils/combineFrameInstance.ts

```diff
@@ -73,26 +67,65 @@ const combineFrameInstance = (frame, instance) => {
         ImagePositionPatientToUse = [position[0], position[1], position[2]];
       }
     }
+    const sharedInstance = createCombinedValue(instance, SharedFunctionalGroupsSequence?.[0]);
+    const newInstance = createCombinedValue(
+      sharedInstance,
+      PerFrameFunctionalGroupsSequence?.[frameNumber]
+    );
 
-    const newInstance = Object.assign(instance, { frameNumber: frameNumber });
-
-    // merge the shared first then the per frame to override
-    [...shared, ...perFrame].forEach(item => {
-      Object.entries(item).forEach(([key, value]) => {
-        newInstance[key] = value;
-      });
+    // Note: do not assign directly to newInstance.ImagePositionPatient
+    // because it will also overwrite the instance.ImagePositionPatient since it
+    // is create via Object.create(parent)
+    Object.defineProperty(newInstance, 'ImagePositionPatient', {
+      value: ImagePositionPatientToUse ?? newInstance.ImagePositionPatient ?? [0, 0, frameNumber],
+      writable: true,
+      enumerable: true,
+      configurable: true,
     });
 
-    // Todo: we should cache this combined instance somewhere, maybe add it
-    // back to the dicomMetaStore so we don't have to do this again.
-    return {
-      ...newInstance,
-      ImagePositionPatient: ImagePositionPatientToUse ??
-        newInstance.ImagePositionPatient ?? [0, 0, frameNumber],
-    };
+    Object.defineProperty(newInstance, 'frameNumber', {
+      value: frameNumber,
+      writable: true,
+      enumerable: true,
+      configurable: true,
+    });
+    return newInstance;
   } else {
     return instance;
   }
 };
 
+function createCombinedValue(parent, functionalGroups) {
+  const newInstance = Object.create(parent);
+  if (!functionalGroups) {
+    return newInstance;
+  }
+  if (functionalGroups._sharedValue) {
+    return functionalGroups._sharedValue;
+  }
+  const shared = functionalGroups
+    ? Object.values(functionalGroups)
+        .filter(Boolean)
+        .map(it => it[0])
+        .filter(it => typeof it === 'object')
+    : [];
+
+  // merge the shared first then the per frame to override
+  [...shared].forEach(item => {
+    if (item.SOPInstanceUID) {
```

Comment:
Should we simply avoid overriding any values in the child?  I know this tests for the SOPInstanceUID, but I would guess other things could get overridden.